src/db_path.cpp.in

```diff
@@ -28,19 +28,54 @@
 #include <miopen/env.hpp>
 #include <miopen/stringutils.hpp>
 #include <miopen/expanduser.hpp>
+#include <miopen/logger.hpp>
+
+#include <boost/filesystem.hpp>
 
 #include <cstdlib>
+#ifdef __linux__
+#include <dlfcn.h>
+#endif
 
 MIOPEN_DECLARE_ENV_VAR(MIOPEN_SYSTEM_DB_PATH)
 MIOPEN_DECLARE_ENV_VAR(MIOPEN_USER_DB_PATH)
 
 namespace miopen {
 
+boost::filesystem::path GetLibPath()
+{
+    boost::filesystem::path path = {""};
+#ifdef __linux__
+    Dl_info info;
+
+    if(dladdr(reinterpret_cast<void*>(miopenCreate), &info) != 0)
+    {
+        path = boost::filesystem::canonical(boost::filesystem::path{info.dli_fname});
+        MIOPEN_LOG_I2(std::string("Lib Path: ") + path.string());
+        if(path.empty())
+            return path;
+
+        path = path.parent_path();
+    }
+    return path;
+#else
+    MIOPEN_THROW(miopenStatusNotImplemented);
+#endif
+}
+
 std::string GetSystemDbPath()
 {
     auto p = GetStringEnv(MIOPEN_SYSTEM_DB_PATH{});
     if(p == nullptr)
+#if MIOPEN_BUILD_DEV
```

Comment:
Why is this `#if` needed if we are no longer defining the `MIOPEN_SYSTEM_DB_PATH` cmake variable?

Also, we can stop configuring this file with cmake and rename it to `db_path.cpp`.

Reply:
I held back on this so that if we do a build with `MIOPEN_BUILD_DEV` set to `true` then the older behavior should be preserved. Keep in mind that the `MIOPEN_SYSTEM_DB_PATH` is set in CMake if Build dev is on.

Reply:
BTW are we aware of any uses of the `MIOPEN_BUILD_DEV` CMake variable? If not, then we can really get rid of it and simplify our code (in a separate PR).

Reply:
@atamazov  I never use it, I use the DEBUG build target for similar effect and install the freshly built MIOpen in the docker to get the same behavior a user would. So I would vote to remove it